src/libsystemd/sd-journal/sd-journal.c

```diff
@@ -1587,6 +1587,29 @@ static int add_any_file(
         assert(j);
         assert(fd >= 0 || path);
 
+        if (path) {
+                f = ordered_hashmap_get(j->files, path);
+                if (f) {
+                        if (stat_inode_same(&f->last_stat, &st)) {
```

Comment:
Hmm? `st` is not initialized just yet? I don't follow this at all?

Reply:
ah yes, you are right. I missed that `st` depends on the `fd`. Basically I am trying to see if there is a way to avoid the close(our_fd) if possible. I am seeing a case where I think we are hitting double close on the same file descriptor.

Trying to see how we can avoid it in the sd-journal code... or maybe it needs to be resolved on the client level.